File: webmvp/src/lib/userDisplay.ts
export function userInitials(name: string): string {
  const words = name.trim().split(/\s+/).filter(Boolean);
  if (words.length === 0) {
    return "?";
  }
  if (words.length === 1) {
    return words[0].slice(0, 2).toUpperCase();
  }
  return (words[0][0] + words[1][0]).toUpperCase();
}

export function userDisplayName(
  displayName: string | null | undefined,
  email: string | null | undefined,
): string {
  if (displayName?.trim()) {
    return displayName.trim();
  }
  if (email) {
    return email.split("@")[0] ?? "Musician";
  }
  return "Guest";
}

File: mobile/lib/domain/user_display.dart
String userInitials(String name) {
  final words = name.trim().split(RegExp(r'\s+')).where((w) => w.isNotEmpty).toList();
  if (words.isEmpty) {
    return '?';
  }
  if (words.length == 1) {
    return words.first.substring(0, words.first.length.clamp(0, 2)).toUpperCase();
  }
  return '${words[0][0]}${words[1][0]}'.toUpperCase();
}

String userDisplayName(String? displayName, String? email) {
  if (displayName != null && displayName.trim().isNotEmpty) {
    return displayName.trim();
  }
  if (email != null && email.isNotEmpty) {
    return email.split('@').first;
  }
  return 'Guest';
}
